tests/timeout_future.rs

```diff
@@ -0,0 +1,27 @@
+#![cfg(feature = "unstable")]
+
+use std::time::Duration;
+
+use async_std::prelude::*;
+use async_std::future;
+use async_std::task;
+
+#[test]
+fn should_timeout() {
+    task::block_on(async {        
+        let fut = future::pending::<()>();
+        let dur = Duration::from_millis(100);
+        let res = fut.timeout(dur).await;
+        assert!(res.is_err());
+    });
+}
+
+#[test]
+fn should_not_timeout() {
+    task::block_on(async {        
+        let fut = future::ready(0);
+        let dur = Duration::from_millis(100);
+        let res = fut.timeout(dur).await;
+        assert!(res.is_ok());
+    });
```

Comment:
It is nice! However, I would like to write this as a **doc test**.
I think it will be easier to use by leaving it as a document.

I think the document comments are as follows
```rust
#[doc = r#"
    Waits for both the future and a timeout, if the timeout completes before
    the future, it returns an TimeoutError.

    # Examples

    ```
    # async_std::task::block_on(async {
    let fut = future::pending::<()>();
    let dur = Duration::from_millis(100);
    assert!(fut.timeout(dur).await.is_err());

    let fut = future::ready(0);
    let dur = Duration::from_millis(100);
    assert!(fut.timeout(dur).await.is_ok());
    # });
    ```
"#]
```

Reply:
Yes, I wrote it after I had pushed earlier, but I was stuck with the format haha